modules/runner/src/main/ts/reporter/MochaReporter.ts

```diff
@@ -0,0 +1,122 @@
+import { LoggedError } from '@ephox/bedrock-common';
+import Promise from '@ephox/wrap-promise-polyfill';
+import { Context, MochaOptions, Runner as MochaRunner, Test } from 'mocha';
+import { Lock } from '../core/Lock';
+import { noop } from '../core/Utils';
+import { Reporter, TestReporter } from './Reporter';
+
+type LoggedError = LoggedError.LoggedError;
+
+export interface MochaReporterOptions {
+  readonly reporter: Reporter;
+  readonly numTests: number;
+  readonly onPass: () => void;
+  readonly onSkip: () => void;
+  readonly onFailure: (err: LoggedError) => void;
+}
+
+const defaultOptions: Partial<MochaReporterOptions> = {
+  onPass: noop,
+  onSkip: noop,
+  onFailure: noop
+};
+
+const isTestEqual = (test1?: Test, test2?: Test) => {
+  if (test1 === undefined || test2 === undefined) {
```

Comment:
boolean expression could be simplified. Not sure how it'd read, though.

Reply:
If both are undefined, should this return true?

Reply:
In reality, neither should ever be `undefined` so I'll update the signature to account for that instead.

Reply:
Changed my mind... as it's potentially going to be more reusable/less error-prone if it accepts both. So have changed it to account for both being passed as undefined as well. I've also made the arguments non-optional, as that wasn't technically correct (we always want 2 arguments passed).